### PR TITLE
Add Maven Disable Artifact, Fingerprinting and Site slicing

### PR DESCRIPTION
**Pull request by mgaunin**
Very useful when many maven jobs have to be updated.
3 new classes developed with MavenIncremental class as model.

### CODE REVIEW

**Comment by aheritier** [comment]
tests failure are not related it seems
Maybe you could have added new tests to cover these features (even if they are "simple") ?

**Comment by mgaunin** [comment]
I thought about it but since they are "only" boolean, I'm not sure about the real added value to add tests...
About failing tests... hummm... the issue comes from hudson.model.Descriptor
```
    protected XmlFile getConfigFile() {
        return new XmlFile(new File(Jenkins.getInstance().getRootDir(), this.getId() + ".xml"));
    }
```
Maybe something is missing at test initialization...

**Comment by mgaunin** [comment]
No more issue with LogfilesizecheckerSlicerTest unit test.
Thought to use Jenkins Mock but not necessary since the merge.
@reviewbybees

**Comment by jmMeessen** [comment]
Had a look at the code (mainly on the changes and not on the rest of the plugin) as a newbie.
- I am wondering about the absence of javaDoc. But it is not used a lot in the rest of the code.
- I am wondering about null check not done in the code. Example when using "item".  Is this a usual practice ? checkNotNull() from Guava is available from the existing pom.xml. At least it would make a failure (nulPointerException) more readable.

Otherwise it looks good to me.

My two cents worth of code reading.

**Comment by mgaunin** [comment]
Thanks Jean-Marc for these feedback.
I prefer to handle your remarks globally but not part of this PR.
Please create an issue for that.

**Comment by dariver** [comment]
I observed some weird code format, maybe you are using tabs instead of spaces.

Addressing the code format, LGTM 👍

File: src/main/java/configurationslicing/maven/MavenDisableArtifact.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        public String getName() {
+            return "Maven Disable automatic artifact archiving";
+        }
+
+        public String getName(MavenModuleSet item) {
+            return item.getFullName();
+        }
+
+        public String getUrl() {
+            return "mavendisableartifact";
```
Comment (oleg-nenashev) [inline src/main/java/configurationslicing/maven/MavenDisableArtifact.java]: commonly Jenkins URLs are camelCase. Would be preferable here IMHO

File: src/main/java/configurationslicing/maven/MavenDisableArtifact.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+public class MavenDisableArtifact extends BooleanSlicer<MavenModuleSet> {
+
+    public MavenDisableArtifact() {
+        super(new MavenDisableArtifactSlicerSpec());
+    }
+    
+    public static class MavenDisableArtifactSlicerSpec implements BooleanSlicerSpec<MavenModuleSet> {
+
+        public String getName() {
+            return "Maven Disable automatic artifact archiving";
```
Comment (oleg-nenashev) [inline src/main/java/configurationslicing/maven/MavenDisableArtifact.java]: Nice 2 have; Keep strings localizable: https://wiki.jenkins-ci.org/display/JENKINS/Internationalization

File: src/main/java/configurationslicing/maven/MavenDisableArtifact.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            return item.getFullName();
+        }
+
+        public String getUrl() {
+            return "mavendisableartifact";
+        }
+
+        @SuppressWarnings("unchecked")
+		public List<MavenModuleSet> getWorkDomain() {
+            return (List) Jenkins.getInstance().getAllItems(MavenModuleSet.class);
```
Comment (oleg-nenashev) [inline src/main/java/configurationslicing/maven/MavenDisableArtifact.java]: Should be no need to cast it here

File: src/main/java/configurationslicing/maven/MavenDisableArtifact.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package configurationslicing.maven;
+
+import configurationslicing.BooleanSlicer;
+import hudson.Extension;
+import hudson.maven.MavenModuleSet;
+import jenkins.model.Jenkins;
+
+import java.util.List;
+
+@Extension
```
Comment (oleg-nenashev) [inline src/main/java/configurationslicing/maven/MavenDisableArtifact.java]: 🐛 Must be marked as optional

File: src/main/java/configurationslicing/maven/MavenDisableFingerprinting.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
+package configurationslicing.maven;
+
+import configurationslicing.BooleanSlicer;
+import hudson.Extension;
+import hudson.maven.MavenModuleSet;
+import jenkins.model.Jenkins;
+
+import java.util.List;
+
+@Extension
```
Comment (oleg-nenashev) [inline src/main/java/configurationslicing/maven/MavenDisableFingerprinting.java]: 🐛 Must be marked as optional

File: src/main/java/configurationslicing/maven/MavenDisableSite.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package configurationslicing.maven;
+
+import configurationslicing.BooleanSlicer;
+import hudson.Extension;
+import hudson.maven.MavenModuleSet;
+import jenkins.model.Jenkins;
+
+import java.util.List;
+
+@Extension
```
Comment (oleg-nenashev) [inline src/main/java/configurationslicing/maven/MavenDisableSite.java]: 🐛 Must be marked as optional

File: src/main/java/configurationslicing/maven/MavenDisableArtifact.java (hunk 2 above)
Comment (mgaunin) [inline src/main/java/configurationslicing/maven/MavenDisableArtifact.java]: I created an issue for that to review all the code of the plugin (see https://issues.jenkins-ci.org/browse/JENKINS-44363)

File: src/main/java/configurationslicing/maven/MavenDisableArtifact.java (hunk 1 above)
Comment (mgaunin) [inline src/main/java/configurationslicing/maven/MavenDisableArtifact.java]: I created an issue for that to review all the code of the plugin (see https://issues.jenkins-ci.org/browse/JENKINS-44367)

Comment (oleg-nenashev) [inline src/main/java/configurationslicing/maven/MavenDisableArtifact.java]: Once you have an URL, you cannot easily get rid of it without compatibility risks. So I would keep the existing links as is if you really do not want to invest significant time into this improvement

File: src/main/java/configurationslicing/maven/MavenDisableArtifact.java (hunk 4 above)
Comment (mgaunin) [inline src/main/java/configurationslicing/maven/MavenDisableArtifact.java]: cf. 'Mark @Extension as optional for Maven Slicers' commit

File: src/main/java/configurationslicing/maven/MavenDisableFingerprinting.java (hunk 5 above)
Comment (mgaunin) [inline src/main/java/configurationslicing/maven/MavenDisableFingerprinting.java]: cf. 'Mark @Extension as optional for Maven Slicers' commit

File: src/main/java/configurationslicing/maven/MavenDisableSite.java (hunk 6 above)
Comment (mgaunin) [inline src/main/java/configurationslicing/maven/MavenDisableSite.java]: cf. 'Mark @Extension as optional for Maven Slicers' commit

File: src/main/java/configurationslicing/maven/MavenDisableArtifact.java (hunk 3 above)
Comment (mgaunin) [inline src/main/java/configurationslicing/maven/MavenDisableArtifact.java]: cf. 'Remove unnecessary cast from Maven Slicers' commit

**Comment by guysoft** [comment]
Hey, adopted this plugin, this PR conflicts now, any change you can fix it so I can merge it? I am not an expert on Java and I don't want to break anything.
There is a new version coming out now that I got this compiling. So it would be geat if you could.

**Comment by guysoft** [comment]
Closing if no reply

**Comment by oleg-nenashev** [comment]
This change still has value, but it indeed requires somebody to finalize it. After 4 years it is unlikely to be relevant to @mgaunin 

I could check whether Mikael's team would be interested in it, but I will need a few days for it

**Comment by guysoft** [comment]
Hey @oleg-nenashev any news? Should I check up or do you know this is still open?

**Pull request closed without merging**